docs/concepts/api-extension/custom-resources.md

```diff
@@ -72,6 +223,318 @@ implementations for your custom resources by writing and deploying your own stan
+   </td>
+   <td>Can I reuse the authorization scheme of the Core API server (e.g. RBAC).  
+   </td>
+  </tr>
+  <tr>
+   <td>Finalizers
+   </td>
+   <td>Can I have resource deletion block on another client doing something.
+   </td>
+  </tr>
+  <tr>
+   <td>Initializers
+   </td>
+   <td>Can I have resource creation block on another client doing something.
+   </td>
+  </tr>
+  <tr>
+   <td>UI/CLI Display
+   </td>
+   <td>Kubectl, dashboard can display objects of your new type, without modification.
+   </td>
+  </tr>
+  <tr>
+   <td>Unset vs Empty
+   </td>
+   <td>Can clients distinguish unset fields from set to 0 or empty-string fields.
+   </td>
+  </tr>
+</table>
+
+
+## Preparing to Install a Custom Resource
+
+There are several points to be aware of before adding a custom resource to your cluster.
+
+### Third Party Code and New Points of Failure
+
+Installing a CRD by itself, (meaning creating a resource of kind CustomResourceDefinition), does not cause any "third party code" to run on your apiserver, and does not add any new point of failure.  However packages (such as Charts) or other installation bundles often include CRDs as well as a Deployment of third-party code that implements the business logic for the new Custom Resource.  
+
+Installing an Aggregated APIserver always involves running a new Deployment.  
+
+
+### Storage 
+
+Custom resources consume storage space in the same way that ConfigMaps do.  Watch out for creating too many, as it may overload your APIserver.  Aggregated API Servers may have their own storage, or may use Custom Resources for their storage, in which case the same warning applies.  
+
+
+### Authentication, Authorization and Auditing
+
+Custom Resources always use the same authentication, authorization, and audit logging as the built-in resources of your API Server.  
+
+If you are using RBAC for authorization, then most RBAC roles will not grant access to the new resources (except the cluster-admin role or any you created with wildcard rules).  You'll need to explicitly grant access to the new resources.  CRDs and Aggregated APIs often come bundled with new role definitions for the types they add.
+
+Aggregated API servers may or may not use the same authentication, authorization and auditing as the primary APIServer. 
+
+
+## Accessing a Custom Resource
+
+Kubernetes [client libraries](https://kubernetes.io/docs/reference/client-libraries/) can be used to access custom resources.  Not all client libraries support custom resources.  The go and python client libraries do.
+
+TODO: talk about how to access the resource from kubectl.  Are binary plugins needed?
+
+TODO: confirm this: When you add a custom resource, you can access it using: your own rest client, a kubernetes dynamic client, a generated client using the open-api spec (is that true?).  When you use api aggregation, you generate a static client from the go-idl. TODO: make glossary and check if these terms are defined on the web site.
```

Comment:
You could generate a static client from go-idl for CRD resources too:

https://blog.openshift.com/kubernetes-deep-dive-code-generation-customresources/